lib/ip_blocklist.js

```diff
@@ -0,0 +1,196 @@
+
+          var parsedData = parseRow(row[0])
+
+          if (parsedData) {
+            if (!newIpsByPrefixLength[parsedData.prefixLength]) {
+              newPrefixLengths.push(parsedData.prefixLength)
+              newIpsByPrefixLength[parsedData.prefixLength] = {}
+            }
+            newIpsByPrefixLength[parsedData.prefixLength][parsedData.baseIp] = true
+          }
+
+          // Every 1000 rows, yield process
+          if (idx % 1000 === 0) {
+            return Promise.fromNode(setImmediate)
+          }
+        })
+      })
+      .then(function () {
+
+        // Update map to new values
+        self.ipsByPrefixLength = newIpsByPrefixLength
+        self.prefixLengths = newPrefixLengths
+
+        var endTime = Date.now()
+        log.info({
+          op: 'fxa.customs.blocklist.load',
+          fileName: self.fileName,
+          fileSize: self.fileSize,
+          loadedIn: (endTime - startTime)
+        })
+      })
+  }
+
+  IPBlocklist.prototype.clear = function () {
+    this.prefixLengths = []
+    this.ipsByPrefixLength = {}
+  }
+
+  IPBlocklist.prototype.contains = function (ipAddress) {
+    var self = this
+
+    var startTime = Date.now()
+    var endTime
+
+    log.info({
+      op: 'fxa.customs.blocklist.check',
+      ip: ipAddress,
+      fileName: self.fileName,
+      fileSize: self.fileSize
+    })
+
+    try {
+      return self.prefixLengths.some(function (prefixLength) {
+        var result = getBaseIp(ipAddress, prefixLength) in self.ipsByPrefixLength[prefixLength]
+
+        // If `hit` log metrics on time taken
+        if (result) {
+          endTime = Date.now()
+          log.info({
+            op: 'fxa.customs.blocklist.hit',
+            ip: ipAddress,
+            fileName: self.fileName,
+            fileSize: self.fileSize,
+            foundIn: (endTime - startTime)
+          })
+        }
+
+        return result
+      })
+    } catch (err) {
+
+      // On any error, fail closed and reject request. These errors could be
+      // malformed ip address, etc
+      log.error({
+        op: 'fxa.customs.blocklist.error',
+        fileName: self.fileName,
+        fileSize: self.fileSize,
+        err: err
+      })
+
+      throw err
+    }
+  }
+
+  IPBlocklist.prototype.pollForUpdates = function () {
+    this.stopPolling()
+    this.pollInterval = setInterval(this.refresh.bind(this), config.ipBlocklist.updatePollInterval * 1000)
+  }
+
+  IPBlocklist.prototype.stopPolling = function () {
+    clearInterval(this.pollInterval)
+  }
+
+  IPBlocklist.prototype.refresh = function () {
+    var self = this
+
+    log.trace({
+      op: 'fxa.customs.blocklist.refreshList',
+      fileName: self.fileName,
+      fileSize: self.fileSize
+    })
+
+    return statFile(self.filePath)
+      .then(function (fileStats) {
+        var mtime = fileStats.mtime
+        if (mtime > self.fileLastModified) {
+          self.fileLastModified = mtime
```

Comment:
IIUC the call to `self.load` will update `self.fileLastModified` for us, so I don't think we need to do it again here?